### PR TITLE
#170947425 Setup unit testing

### PR DESCRIPTION
**Pull request by blaise82**
#### What does this PR do?
ch(unity testing): Setup unit testing

#### Description of Task to be completed?
- install mocha/chai
- write one test

#### How should this be manually tested?
- clone this repo
- run npm install
- run npm test

#### Screenshots (if appropriate)
Screen Shot 2020-02-05 at 16 42 27

[finishes #170947425]

### CODE REVIEW

File: src/test/server.test.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import chai, { expect } from 'chai';
```
[the remark this answers is not in the document]
Comment (AbdoulNuru) [reply src/test/server.test.js:1]: @blaise82 @izzett222 what would be the cause of this

**Review by hakoemmy — CHANGES_REQUESTED** [review]
Good job @blaise82,  Why did you write **ch(unity testing): Setup unit testing** for ####What does this PR do? I think you should follow Andela's best practices.

**Review by AbdoulNuru — APPROVED** [review]

**Review by hakoemmy — CHANGES_REQUESTED** [review]
@blaise82 Take a look at your committing approach: 

ch(unity testing): Setup unit testing …
- install mocha/chai
- write one test
it doesn't adhere [Andela bestpractices](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices) 

Please rename your commit message to look like:

chore(coveralls):add coveralls yml
[Maintains #153742460]

**Comment by blaise82** [comment]
> @blaise82 Take a look at your committing approach:
> 
> ch(unity testing): Setup unit testing …
> 
> * install mocha/chai
> * write one test
>   it doesn't adhere [Andela bestpractices](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices)
> 
> Please rename your commit message to look like:
> 
> chore(coveralls):add coveralls yml
> [Maintains #153742460]

I am not working on coveralls 
`<type>(<scope>): <subject>``<BLANK LINE> <body> <BLANK LINE> <footer>`
```
<type>chore
(<scope>):unity testing
 <subject>: Setup unit testing.
............
```

**Comment by hakoemmy** [comment]
> > @blaise82 Take a look at your committing approach:
> > ch(unity testing): Setup unit testing …
> > 
> > * install mocha/chai
> > * write one test
> >   it doesn't adhere [Andela bestpractices](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices)
> > 
> > Please rename your commit message to look like:
> > chore(coveralls):add coveralls yml
> > [Maintains #153742460]
> 
> I am not working on coveralls
> `` <type>(<scope>): <subject>``<BLANK LINE> <body> <BLANK LINE> <footer> ``
> 
> ```
> <type>chore
> (<scope>):unity testing
>  <subject>: Setup unit testing.
> ............
> ```

@blaise82 I was showing you an example of a good commit message

**Comment by hakoemmy** [comment]
> > > @blaise82 Take a look at your committing approach:
> > > ch(unity testing): Setup unit testing …
> > > 
> > > * install mocha/chai
> > > * write one test
> > >   it doesn't adhere [Andela bestpractices](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices)
> > > 
> > > Please rename your commit message to look like:
> > > chore(coveralls):add coveralls yml
> > > [Maintains #153742460]
> > 
> > 
> > I am not working on coveralls
> > ` <type>(<scope>): <subject>`    ``
> > ```
> > <type>chore
> > (<scope>):unity testing
> >  <subject>: Setup unit testing.
> > ............
> > ```
> 
> @blaise82 I was showing you an example of a good commit message

**Review by RusimbiPatrick — APPROVED** [review]

**Review by AbdoulNuru — APPROVED** [review]

File: src/index.js [hunk 2]
```diff
@@ -1,96 +1,55 @@
...
-
-if (isProduction) {
-    mongoose.connect(process.env.MONGODB_URI);
-} else {
-    mongoose.connect("mongodb://localhost/conduit");
-    mongoose.set("debug", true);
-}
-
-require("./models/User");
-
```
Comment (izzett222) [inline src/index.js]: hey @blaise82 , aren't those lines of code important for the project?

Comment (blaise82) [reply src/index.js]: @izzett222 not real 
for example 
`if (isProduction) {
    mongoose.connect(process.env.MONGODB_URI);
} else {
    mongoose.connect("mongodb://localhost/conduit");
    mongoose.set("debug", true);
}`

is used to check if we are connected to MONGODB 
and we are not using MONGO

**Comment by blaise82** [comment]
> > > @blaise82 Take a look at your committing approach:
> > > ch(unity testing): Setup unit testing …
> > > 
> > > * install mocha/chai
> > > * write one test
> > >   it doesn't adhere [Andela bestpractices](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices)
> > > 
> > > Please rename your commit message to look like:
> > > chore(coveralls):add coveralls yml
> > > [Maintains #153742460]
> > 
> > 
> > I am not working on coveralls
> > ` <type>(<scope>): <subject>`    ``
> > ```
> > <type>chore
> > (<scope>):unity testing
> >  <subject>: Setup unit testing.
> > ............
> > ```
> 
> @blaise82 I was showing you an example of a good commit message

ohhh greate

**Review by hakoemmy — APPROVED** [review]

**Review by izzett222 — APPROVED** [review]

File: src/index.js [hunk 3]
```diff
@@ -1,96 +1,55 @@
...
-const isProduction = process.env.NODE_ENV === "production";
+const isProduction = process.env.NODE_ENV === 'production';
 
 // Create global app object
 const app = express();
 
 app.use(cors());
 
-// Normal express config defaults
-app.use(require("morgan")("dev"));
```
Comment (kcharles52) [inline src/index.js:21]: we need this middleware, please put it back

File: src/index.js [hunk 4]
```diff
@@ -1,96 +1,55 @@
...
-    http = require("http"),
-    path = require("path"),
-    methods = require("methods"),
-    express = require("express"),
-    bodyParser = require("body-parser"),
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
```
Comment (kcharles52) [inline src/index.js:11]: we need the following modules
cors, session, passport

**Review by kcharles52 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 4 above)
Comment (blaise82) [reply src/index.js:11]: Mongoose too  ` mongoose = require("mongoose");`

Comment (kcharles52) [reply src/index.js:11]: No, we don't need mongoose, we are going to use sequalize

**Review by kcharles52 — APPROVED** [review]

File: src/test/server.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (blaise82) [reply src/test/server.test.js:1]: @AbdoulNuru please see if the following link may help you solve the problem 


https://github.com/AtomLinter/linter-eslint/issues/462

**Pull request merged**